Purchases/Misc/Deprecations.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Deprecations.swift
```

Comment:
I kinda feel like `Obsoletions.swift` might be more appropriate as a file name, since there are still deprecated methods that won't make it to this file, but all obsoleted methods should.